nexus/db-queries/src/db/datastore/deployment/external_networking.rs

```diff
@@ -281,7 +282,21 @@ impl DataStore {
         log: &Logger,
     ) -> Result<bool, Error> {
         // See the comment in is_external_ip_already_allocated().
-        if cfg!(any(test, feature = "testing")) && nic.ip.is_loopback() {
+        //
+        // TODO-completeness: Ensure this works for dual-stack Omicron service
+        // zone NICs. See https://github.com/oxidecomputer/omicron/issues/9313.
+        if cfg!(any(test, feature = "testing"))
+            && nic
+                .ip_config
+                .ipv4_addr()
+                .map(|ip| ip.is_loopback())
+                .unwrap_or(false)
+            && nic
```

Comment:
This means that if there is an IPv4 loopback address, but no IPv6 address at all, we return false. Is that the intent?

Reply:
Yes, but this is all behind a `cfg!(any(feature, "testing"))` guard. @jgallagher Does this look right? I tried to keep the original intent, handling both IP stacks.